src/utils.ts

```diff
@@ -533,6 +548,14 @@ export namespace Utils {
       );
     }
 
+    let clientSigningKey;
+    for (const op of tx.operations) {
+      if (op.type === "manageData" && op.name === "client_domain") {
+        clientSigningKey = op.source;
+        break;
+      }
+    }
```

Comment:
What's the expected behavior if there's more than one of these? If it's an error, we can do some finagling here, e.g.:

```suggestion
    let clientSigningKey;
    const clientSigningKeyCandidates = tx.operations.filter(
        op => op.type === "manageData" && op.name === "client_domain");
    switch (len(clientSigningKeyCandidates)) {
        case 0: break;
        case 1: 
            clientSigningKey = clientSigningKeyCandidates[0]; 
            break;
        default:
            throw Error("Found more than one signing key???");
      }
```

Reply:
It is not defined in the protocol, but there should only be one of these operations. If there is more than one, we can throw. I'll update the code with your changes 👍